qiskit/circuit/classical/expr/visitors.py

```diff
@@ -297,4 +299,7 @@ def is_lvalue(node: expr.Expr, /) -> bool:
             >>> expr.is_lvalue(expr.bit_and(a, b))
             False
     """
+    if node.const:
+        # A constant expression is _never_ an lvalue.
+        return False
     return node.accept(_IS_LVALUE)
```

Comment:
Might be cleaner just to have the `_IS_VALUE` visitor handle this - adding it here feels like splitting the logic a little bit, and it shouldn't really save anything the short-circuit, since it'd happen immediately on the first visit anyway.

Maybe indicative that the visitor is missing some logic: should it be handling `visit_var` as `return not var.const`?

Reply:
I actually don't think we depend on this at all, anyway. I can just remove it for now. I think we decided that `Var` cannot be `const`, correct?

It could eventually be short-circuiting though, like if we ever allow durations inside arrays (I think QASM allows this).

Reply:
Ah alright, I see what you're getting at. Removed it in 49f2af835254b1b51cd18166fbefa773c46f56c0